src/unittest/test.h

```diff
@@ -113,6 +99,27 @@ class TestBase {
 	u32 num_tests_failed;
 	u32 num_tests_run;
 
+	template <typename Test>
+	void runTest(const char *name, Test &&test) {
+		u64 t1 = porting::getTimeMs();
+		try {
+			test();
```

Comment:
I don’t understand how does your suggestion relate to the code.
However, if you don’t like the template I can make `test` an `std::function<void()>` instead, it would work equally well. And move the function to the cpp as it’d be possible then.

Reply:
Do I remember correctly that templates are just fancy macros? They are copied for each typename. Thus I would welcome to keep this template as small as possible by either
 *  catch and forward the exception to a generic helper function to handle it  .. or ..
 * wrap the call into a `std::function` as you suggested.

In both cases the function (or large parts of it) could be moved to the source file (cpp), which is what I'd also like to see.

Reply:
Updated. But I don’t understand your concern. Is that resulting binary size?